controller/userController.js

```diff
@@ -0,0 +1,45 @@
+var user = require("../model/userModel");
+const bcrypt = require("bcrypt");
+
+module.exports = {
+  signUp: async (req, res) => {
+    if (!req.body.email) {
+      res.send({ responseCode: 404, responseMessage: "email is required !" });
+    }
+    var query = { email: req.body.email };
+    const result = await user.findOne(query);
+    try {
+      if (result) {
+        if (result.email == req.body.email) {
+          res.json({
+            responseCode: 409,
+            responseMessage: "Email already exist.",
+          });
+        }
+      } else {
+        const saltRounds = 10;
+        bcrypt.hash(req.body.password, saltRounds, function (err, hash) {
```

Comment:
here also you can use async await check proper doc for each module you are using